DependencyInjection/Configuration.php

```diff
@@ -97,7 +97,7 @@ private function addDbalSection(ArrayNodeDefinition $node)
                             ->end()
                             ->children()
                                 ->scalarNode('class')->isRequired()->end()
-                                ->booleanNode('commented')->defaultTrue()->end()
+                                ->booleanNode('commented')->defaultFalse()->end()
```

Comment:
I suggest a BC way:

- trigger a deprecation whenever the field is configured explicitly (to allow removal), by using `beforeNormalization`
- if the field is configured as commented (either explicitly or due to the default), the DI extension should check whether the type class overwrites the `requiresSQLCommentHint` method:
  - If no, it should trigger a deprecation saying that types should be marked as commented in the type itself.
  - If yes, it should remove this type from the list of commented types passed to the bundle boot method (actually marking it as not commented in the parameter). Types will still be registered as commented in DBAL due to the type class itself.